File: 0122_best_time_to_buy_and_sell_stock_II/max_profit.go
package main

/**
 * Time: O(n)
 * Space: O(1)
 */
func maxProfit(prices []int) int {
	profits := 0
	for i := 1; i < len(prices); i++ {
		if prices[i] > prices[i-1] {
			profits += prices[i] - prices[i-1]
		}
	}
	return profits
}

func main() {

}

File: 0122_best_time_to_buy_and_sell_stock_II/max_profit.js
/**
 * Time:    O(n)
 * Space:   O(1)
 */
const maxProfit = (prices) => {
  return prices.reduce((max, el, i) => {
    if (el > prices[i - 1]) max += el - prices[i - 1]
    return max
  }, 0)
}
